base_tier_validation/models/res_users.py

```diff
@@ -13,19 +13,22 @@ class Users(models.Model):
     def review_user_count(self):
         user_reviews = {}
         to_review_docs = {}
-        for review in self.env.user.review_ids.filtered(
-            lambda r: r.status == "pending"
-        ):
+        reviews = self.env["tier.review"].search(
+            [
+                ("id", "in", self.env.user.review_ids.ids),
+                ("status", "=", "pending"),
+                ("can_review", "=", True),
+            ]
+        )
+        for review in reviews:
             record = (
                 review.env[review.model]
                 .with_user(self.env.user)
                 .search([("id", "=", review.res_id)])
             )
-            if not record:
+            if not record or record.rejected or not record.can_review:
```

Comment:
is this something you found problmeatic? I mean being able to validate previously rejected reviews?

Reply:
Yes, I found this problematic with the counter. Even when the document is rejected, but the review counter still increase because it count the rest of the tiers.